ebpf/aya-log-ebpf-macros/src/expand.rs

```diff
@@ -92,6 +92,9 @@ pub(crate) fn log(args: LogArgs, level: Option<TokenStream>) -> Result<TokenStre
             let write_args = quote! {{
                 use ::aya_log_ebpf::WriteToBuf;
                 Ok::<_, ()>(record_len) #( .and_then(|record_len| {
+                    if record_len + core::mem::size_of_val(& { #formatting_exprs } ) >= buf.buf.len() {
```

Comment:
I think this should be just `if record_len >= buf.buf.len()` - `size_of_val` isn't necessarily the size it'll take to format `#formatting_exprs`and the bounds for that are checked inside `write()` anyway.

EDIT: after reviewing the rest, I think `if record_len >= buf.buf.len()` here is the only check needed?